src/libman/Commands/BaseCommand.cs

```diff
@@ -0,0 +1,174 @@
+            {
+                if (RootDir.HasValue())
+                {
+                    HostEnvironment.UpdateWorkingDirectory(GetProjectDirectory());
+                }
+
+                InitializeDependencies();
+
+                return await ExecuteInternalAsync();
+            });
+            Parent = parent;
+
+            return this;
+        }
+
+        private void InitializeDependencies()
+        {
+            ManifestDependencies = new Dependencies(HostEnvironment);
+        }
+
+        private string GetProjectDirectory()
+        {
+            string projectPath = RootDir.Value();
+            if (!Path.IsPathRooted(projectPath))
+            {
+                projectPath = Path.Combine(Directory.GetCurrentDirectory(), projectPath);
+            }
+
+            if (File.Exists(projectPath))
+            {
+                projectPath = Path.GetDirectoryName(projectPath);
+            }
+
+            if (!Directory.Exists(projectPath))
+            {
+                throw new DirectoryNotFoundException(string.Format(Resources.DirectoryNotFoundMessage, projectPath));
+            }
+
+            return projectPath;
+        }
+
+        protected virtual Task<int> ExecuteInternalAsync()
+        {
+            ShowHelp();
+
+            return Task.FromResult(0);
+        }
+
+        /// <summary>
+        /// Provides the help text for the command.
+        /// </summary>
+        /// <param name="commandName"></param>
+        /// <returns></returns>
+        public override string GetHelpText(string commandName = null)
+        {
+            string baseHelp = base.GetHelpText(commandName).Replace("dotnet libman", "libman");
+            var help = new StringBuilder(baseHelp);
+
+            if (!string.IsNullOrWhiteSpace(Remarks))
+            {
+                help.Append($"{Environment.NewLine}{Resources.RemarksHeader}{Environment.NewLine}");
+                help.Append($"{Remarks}{Environment.NewLine}");
+            }
+
+            if (!string.IsNullOrWhiteSpace(Examples))
+            {
+                help.Append($"{Environment.NewLine}{Resources.ExamplesHeader}{Environment.NewLine}");
+                help.Append($"{Examples}{Environment.NewLine}");
+            }
+
+            return help.ToString();
+        }
+
+        protected async Task<Manifest> GetManifestAsync(bool createIfNotExists = false)
+        {
+            if (!File.Exists(Settings.ManifestFileName) && !createIfNotExists)
+            {
+                throw new InvalidOperationException(string.Format(Resources.LibmanJsonNotFound, Settings.ManifestFileName));
+            }
+
+            IDependencies dependencies = ManifestDependencies;
+            Manifest manifest = await Manifest.FromFileAsync(Settings.ManifestFileName, dependencies, CancellationToken.None);
+
+            if (manifest == null)
+            {
+                throw new InvalidOperationException(Resources.FixManifestFile);
+            }
```

Comment:
We use PredefinedErrors.ManifestMalformed() for the vsix if manifest is null. Should we use the same here?